### PR TITLE
Adds Testing section to the README

### LINKED ISSUE #666: [docs] add something about grunt or unit tests to readme?
@rwaldron is there a proper process contributors should follow with respect to Grunt or adding unit tests when creating a new feature on the CLI? If so, this would be great to have in the readme.

### PR DESCRIPTION
**Pull request by HipsterBrown**
Fixes #666

### CODE REVIEW

**Comment by rwaldron** [comment]
Helllllll Yeah

File: README.md [hunk 1]
```diff
@@ -56,11 +58,70 @@ compinit
...
+- [nodeunit](https://github.com/caolan/nodeunit): used to create and run unit tests
+
+All of these tasks can be run by entering `npm test` (an alias for `grunt test` without needing to have the `grunt-cli` installed globally) into your command line. Be sure to run this command before pushing new code so JSHint and style errors can be caught as soon as possible. 
+
+### Writing Tests
+Check out the [nodeunit documentation](https://github.com/caolan/nodeunit#usage) for general usage of the library. For this project, there is a global test setup file (found under [`test/common/bootstrap.js`(https://github.com/tessel/t2-cli/blob/master/test/common/bootstrap.js)) for including any dependencies needed to run the various test modules, as well as two simulator modules, [`RemoteProcessSimulator`](https://github.com/tessel/t2-cli/blob/master/test/common/remote-process-simulator.js) and [`TesselSimulator`](https://github.com/tessel/t2-cli/blob/master/test/common/tessel-simulator.js), for simulating command line and hardware interactions in the tests.
```
Comment (wyze) [inline README.md]: Missing closing `]` in ```[`test/common/bootstrap.js`(```.

**Comment by HipsterBrown** [comment]
That was quick! 😄 Thanks @rwaldron!

File: README.md (hunk 1 above)
Comment (HipsterBrown) [inline README.md]: Oh nice catch @wyze. I'll push a followup fix. 👍

Comment (rwaldron) [inline README.md]: I missed that. @HipsterBrown just push your fix to master

Comment (wyze) [inline README.md]: Oops. Then disregard my PR I just sent, lol.

**Pull request merged**